src/server/services/license/license.repository.ts

```diff
@@ -0,0 +1,194 @@
+    case SortableKeys.NAME:
+      return licenses.name;
+    case SortableKeys.DESCRIPTION:
+      return licenses.description;
+    default:
+      return licenses.id;
+  }
+};
+const generateSortFunction = (
+  key?: SortableKeys | string,
+  value?: string,
+): SQL => {
+  if (
+    value &&
+    Object.keys(SortableKeys).findIndex((sortableKey) => sortableKey === key) >
+      -1
+  ) {
+    if (!["asc", "desc"].includes(value)) {
+      throw new TRPCError({
+        code: "BAD_REQUEST",
+        message: "Invalid sort direction",
+      });
+    }
+
+    return value === "asc"
+      ? asc(mapKeyToColumn(key as SortableKeys))
+      : desc(mapKeyToColumn(key as SortableKeys));
+  }
+
+  // If there is no 'value' or key is not a member of SortableKeys
+  return asc(licenses.name);
+};
+const prepareOrderBy = (sort?: string | string[]): SQL[] => {
+  if (!sort) {
+    return [asc(licenses.type)];
+  }
+
+  if (typeof sort === "string") {
+    const [key, value] = sort.split(":");
+
+    return [generateSortFunction(key, value)];
+  }
+
+  const sorts = [];
+  for (const _sort of sort) {
+    const [key, value] = _sort.split(":");
+    const sortValue = generateSortFunction(key, value);
+    if (sortValue) {
+      sorts.push(sortValue);
+    }
+  }
+
+  return sorts;
+};
+const mapFilterableKeyToConditional = (
+  key: FilterableKeys,
+  value: string,
+): SQL | undefined => {
+  if (key === FilterableKeys.DESCRIPTION || key)
+    return ilike(mapKeyToColumn(key), `${value}%`);
+
+  if (key === FilterableKeys.TYPE || key === FilterableKeys.NAME)
+    return eq(mapKeyToColumn(key as FilterableKeys), value);
+
+  return undefined;
+};
+const prepareWhere = (
+  filter: Record<string, string> | undefined,
+): SQL | undefined => {
+  if (!filter) return undefined;
+
+  if (Object.keys(filter).length === 1) {
+    const [key, value] = Object.entries(filter)[0] ?? ["", ""];
+    if (!Object.values(FilterableKeys).includes(key as FilterableKeys)) {
+      return undefined;
+    }
+
+    return mapFilterableKeyToConditional(key as FilterableKeys, value);
+  }
+
+  const conditionals = [];
+  for (const [key, value] of Object.entries(filter)) {
+    if (!Object.values(FilterableKeys).includes(key as FilterableKeys)) {
+      continue;
+    }
+
+    conditionals.push(
+      mapFilterableKeyToConditional(key as FilterableKeys, value),
+    );
+  }
+
+  return and(...conditionals);
+};
+const licenseRepository = {
+  find: async (data: FindLicenseQuery) => {
+    const { page, limit, sort, filter } = data;
+    const orderBy = prepareOrderBy(sort);
+    const where = prepareWhere(filter);
+
+    const { totalCount, returnData } = await db.transaction(
+      async (
+        tx,
+      ): Promise<{
+        totalCount: number;
+        returnData: FindLicenseReturnDTO[];
+      }> => {
+        const [totalCount] = await tx
+          .select({ count: count() })
+          .from(licenses)
+          //.leftJoin(profiles, eq(users.id, profiles.userId))
```

Comment:
Please remove the commented code because it is not relevant for this repository and it can cause just a confusion for somebody who does not know that it is from the users repository.